src/Field/Base/InputData/FormModelInputData.php

```diff
@@ -22,6 +22,11 @@ final class FormModelInputData implements InputDataInterface
      */
     private ?iterable $validationRules = null;
 
+    private bool $useCustomName = false;
+    private ?string $customName = null;
```

Comment:
```suggestion
    private ?string $customName = null;
```

Looks like `$useCustomName` is not needed.

Use empty string `''` when set empty `$customName` and check `$customName === null` to get if `$customName` was set.
`->withName('')` instead of `->withName(null)`

The same for `$customValue`

Reply:
Empty (`''`) and `null` values are different:
- empty: `<input name>`
- `null`: `<input>`

Also I have redone everything. I removed new methods from `InputDataInterface` and created independent `InputDataWithCustomNameAndValueTrait` trait. I think this is more correct solution.